packages/psss/src/views/Tabs/ProfileTabView.js

```diff
@@ -0,0 +1,149 @@
+
+  const onSubmit = handleSubmit(async ({ firstName, lastName, role, employer }) => {
+    setIsLoading(true);
+    setErrorMessage(null);
+    try {
+      await onUpdateProfile(user.id, {
+        first_name: firstName,
+        last_name: lastName,
+        position: role,
+        employer,
+      });
+      setIsLoading(false);
+      setSuccessMessage('Profile successfully updated.');
+    } catch (error) {
+      setErrorMessage(error.message);
+      setIsLoading(false);
+    }
+  });
+
+  const { firstName, lastName, position, employer } = user;
+
+  return (
+    <Main>
+      <Container>
+        <form onSubmit={onSubmit} noValidate>
+          {errorMessage && <ErrorMessage>{errorMessage}</ErrorMessage>}
+          {successMessage && <SuccessMessage>{successMessage}</SuccessMessage>}
+          <TextField
+            label="First Name"
+            name="firstName"
+            required
+            error={!!errors.firstName}
+            helperText={errors.firstName && errors.firstName.message}
+            defaultValue={firstName}
+            inputRef={register({
+              required: 'Required',
+            })}
+          />
+          <TextField
+            label="Last Name"
+            name="lastName"
+            required
+            error={!!errors.lastName}
+            helperText={errors.lastName && errors.lastName.message}
+            defaultValue={lastName}
+            inputRef={register({
+              required: 'Required',
+            })}
+          />
+          <TextField
+            label="Employer"
+            name="employer"
+            required
+            error={!!errors.employer}
+            helperText={errors.employer && errors.employer.message}
+            defaultValue={employer}
+            inputRef={register({
+              required: 'Required',
+            })}
+          />
+          <TextField
+            label="Role"
+            name="role"
+            required
+            error={!!errors.role}
+            helperText={errors.role && errors.role.message}
+            defaultValue={position}
+            inputRef={register({
+              required: 'Required',
+            })}
+          />
+          <StyledButton type="submit" fullWidth isLoading={isLoading}>
+            Update Profile
+          </StyledButton>
+        </form>
+      </Container>
+    </Main>
+  );
+});
+
+ProfileTabViewComponent.propTypes = {
+  onUpdateProfile: PropTypes.func.isRequired,
+  user: PropTypes.PropTypes.shape({
+    id: PropTypes.string,
+    name: PropTypes.string,
+    firstName: PropTypes.string,
+    lastName: PropTypes.string,
+    employer: PropTypes.string,
+    position: PropTypes.string,
+  }),
+};
+
+ProfileTabViewComponent.defaultProps = {
+  user: null,
+};
+
+const mapStateToProps = state => ({
+  user: getCurrentUser(state),
+});
+
+const mapDispatchToProps = dispatch => ({
+  onUpdateProfile: (id, payload) => dispatch(updateProfile(id, payload)),
```

Comment:
```suggestion
  onUpdateProfile: payload => dispatch(updateProfile(payload)),
```
If we change to using `/me`